Improve default max frequency implementation to reduce repetition.

Random [test failure on OpenBSD](https://github.com/oshi/oshi/actions/runs/5053077466/jobs/9066567004):
```
  Error:  Failures (1):
  Error:    JUnit Jupiter:CentralProcessorTest:testFrequencies()
  Error:      MethodSource [className = 'oshi.hardware.CentralProcessorTest', methodName = 'testFrequencies', methodParameterTypes = '']
  Error:      => java.lang.AssertionError: Logical processor frequency should be at most it's max frequency
  Error:  Expected: is a value less than or equal to <3517000000L>
  Error:       but: <8554633400317716800L> was greater than <3517000000L>
  Error:         org.hamcrest@2.2/org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
  Error:         com.github.oshi@6.4.3-SNAPSHOT/oshi.hardware.CentralProcessorTest.testFrequencies(CentralProcessorTest.java:68)
```

Interestingly, the max frequency just calls the same code as current frequency:
https://github.com/oshi/oshi/blob/4dd1d2909504265c76180a6ab272b8fe83f87ca3/oshi-core/src/main/java/oshi/hardware/platform/unix/openbsd/OpenBsdCentralProcessor.java#L93-L106

These should always be identical except for random sysctl return values.

This does uncover an improvement we could make, though.  The superclass abstract method requires subclasses to implement:
https://github.com/oshi/oshi/blob/4dd1d2909504265c76180a6ab272b8fe83f87ca3/oshi-core/src/main/java/oshi/hardware/common/AbstractCentralProcessor.java#L120

But we could just (by default) iterate over the current frequencies and pick a max, thus allowing subclasses like `OpenBsdCentralProcessor` (and any others that duplicate calls like this) to omit the max implementation, only making that sysctl call once.

--- a/oshi-core/src/main/java/oshi/hardware/common/AbstractCentralProcessor.java
+++ b/oshi-core/src/main/java/oshi/hardware/common/AbstractCentralProcessor.java
@@ -1,5 +1,5 @@
 /*
- * Copyright 2016-2022 The OSHI Project Contributors
+ * Copyright 2016-2023 The OSHI Project Contributors
  * SPDX-License-Identifier: MIT
  */
 package oshi.hardware.common;
@@ -117,7 +117,9 @@ public abstract class AbstractCentralProcessor implements CentralProcessor {
      *
      * @return The max frequency.
      */
-    protected abstract long queryMaxFreq();
+    protected long queryMaxFreq() {
+        return Arrays.stream(getCurrentFreq()).max().orElse(-1L);
+    }
 
     @Override
     public long[] getCurrentFreq() {

--- a/oshi-core/src/main/java/oshi/hardware/platform/unix/openbsd/OpenBsdCentralProcessor.java
+++ b/oshi-core/src/main/java/oshi/hardware/platform/unix/openbsd/OpenBsdCentralProcessor.java
@@ -1,5 +1,5 @@
 /*
- * Copyright 2021-2022 The OSHI Project Contributors
+ * Copyright 2021-2023 The OSHI Project Contributors
  * SPDX-License-Identifier: MIT
  */
 package oshi.hardware.platform.unix.openbsd;
@@ -88,11 +88,6 @@ public class OpenBsdCentralProcessor extends AbstractCentralProcessor {
         // stepping is bits 3:0
         int stepping = cpuid & 0xf;
         return new Triplet<>(family, model, stepping);
-    }
-
-    @Override
-    protected long queryMaxFreq() {
-        return queryCurrentFreq()[0];
     }
 
     @Override

